src/crawlee/base_storage_client/_base_key_value_store_client.py

```diff
@@ -114,3 +115,16 @@ async def delete_record(self, key: str) -> None:
         Args:
             key: The key of the record which to delete
         """
+
+    @abstractmethod
+    def get_public_url(self, key: str, name: str | None, configuration: Configuration) -> str:
+        """Get the public URL for the given key.
+
+        Args:
+            name: Identifier for the key value store
```

Comment:
This isn't necessary - the implementation should already know the name of the key-value store it is handling.